### PR TITLE
feat(freespace_planner, mission_planner, scenario_selector): check if route handler is ready

### PR DESCRIPTION
**Pull request by shmpwk**
Signed-off-by: Shumpei Wakabayashi <shumpei.wakabayashi@tier4.jp>

## Description

Check if route handler is ready.
If not ready, freespace_planner, mission_planner, scenario_selector do not receive route.

Related PR: 
https://github.com/autowarefoundation/autoware.universe/pull/1149
https://github.com/autowarefoundation/autoware.universe/pull/1153

## Pre-review checklist for the PR author

The PR author **must** check the checkboxes below when creating the PR.

- [x] I've confirmed the [contribution guidelines].
- [x] The PR follows the [pull request guidelines].

## In-review checklist for the PR reviewers

The PR reviewers **must** check the checkboxes below before approval.

- [ ] The PR follows the [pull request guidelines].

## Post-review checklist for the PR author

The PR author **must** check the checkboxes below before merging.

- [ ] There are no open discussions or they are tracked via tickets.

After all checkboxes are checked, anyone who has write access can merge the PR.

[contribution guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/
[pull request guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/pull-request-guidelines/

### CODE REVIEW

File: planning/mission_planner/src/mission_planner_lanelet2/mission_planner_lanelet2.cpp [hunk 1]
```diff
@@ -153,47 +153,49 @@ bool MissionPlannerLanelet2::isRoutingGraphReady() const { return is_graph_ready
 void MissionPlannerLanelet2::visualizeRoute(
   const autoware_auto_planning_msgs::msg::HADMapRoute & route) const
 {
-  lanelet::ConstLanelets route_lanelets;
-  lanelet::ConstLanelets end_lanelets;
-  lanelet::ConstLanelets normal_lanelets;
-  lanelet::ConstLanelets goal_lanelets;
-
```
Comment (taikitanaka3) [inline planning/mission_planner/src/mission_planner_lanelet2/mission_planner_lanelet2.cpp:160]: @shmpwk 
If you add below you can make 50 line diff to one line if Im not wrong
if(!route_handler_.isHandlerReady()) return;

Comment (shmpwk) [reply planning/mission_planner/src/mission_planner_lanelet2/mission_planner_lanelet2.cpp]: You're right thank you.
https://github.com/autowarefoundation/autoware.universe/pull/1155/commits/eb164dd7b1454c8e669b8ef07aaddad5f6923c30

File: planning/scenario_selector/src/scenario_selector_node/scenario_selector_node.cpp [hunk 2]
```diff
@@ -215,13 +215,17 @@ void ScenarioSelectorNode::onMap(
...
+  route_handler_ = std::make_shared<route_handler::RouteHandler>(*msg);
 }
 
 void ScenarioSelectorNode::onRoute(
   const autoware_auto_planning_msgs::msg::HADMapRoute::ConstSharedPtr msg)
 {
-  route_ = msg;
-  current_scenario_ = tier4_planning_msgs::msg::Scenario::EMPTY;
+  route_handler_->setRoute(*msg);
+  if (route_handler_->isHandlerReady()) {
```
Comment (taikitanaka3) [inline planning/scenario_selector/src/scenario_selector_node/scenario_selector_node.cpp:225]: please simplify this fix to if() return to make less diff

File: planning/freespace_planner/src/freespace_planner/freespace_planner_node.cpp [hunk 3]
```diff
@@ -307,12 +307,14 @@ void FreespacePlannerNode::getAstarParam()
 
 void FreespacePlannerNode::onRoute(const HADMapRoute::ConstSharedPtr msg)
 {
-  route_ = msg;
+  if (scenario_ != nullptr) {
```
Comment (taikitanaka3) [inline planning/freespace_planner/src/freespace_planner/freespace_planner_node.cpp:310]: please simplify this diff

**Review by taikitanaka3 — APPROVED** [review]
work fine for nominal case
LGTM
image

**Pull request merged**